From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tariq Bashir <tbashir@brightsign.biz>
Date: Thu, 30 Apr 2026 11:05:57 +0100
Subject: Brightsign add back LookupServerBasicAuthCredentials OS-18647

Add back network::mojom::NetworkContext::LookupServerBasicAuthCredentials
as we use this for MediaResourceGetterImpl. This patch partially
reverses:
https://chromium-review.googlesource.com/c/chromium/src/+/6881109

diff --git a/services/network/network_context.cc b/services/network/network_context.cc
index fd0ef3275b205e94e3fa1914d4e64023cdf702dd..159225eec24ba75a7ea0fb72bd622d2836375a05 100644
--- a/services/network/network_context.cc
+++ b/services/network/network_context.cc
@@ -2778,6 +2778,24 @@ void NetworkContext::SetCorsNonWildcardRequestHeadersSupport(bool value) {
       cors::NonWildcardRequestHeadersSupport(value);
 }
 
+void NetworkContext::LookupServerBasicAuthCredentials(
+    const GURL& url,
+    const net::NetworkAnonymizationKey& network_anonymization_key,
+    LookupServerBasicAuthCredentialsCallback callback) {
+  net::HttpAuthCache* http_auth_cache =
+      url_request_context_->http_transaction_factory()
+          ->GetSession()
+          ->http_auth_cache();
+  net::HttpAuthCache::Entry* entry = http_auth_cache->LookupByPath(
+      url::SchemeHostPort(url), net::HttpAuth::AUTH_SERVER,
+      network_anonymization_key, std::string(url.path()));
+  if (entry && entry->scheme() == net::HttpAuth::AUTH_SCHEME_BASIC) {
+    std::move(callback).Run(entry->credentials());
+  } else {
+    std::move(callback).Run(std::nullopt);
+  }
+}
+
 #if BUILDFLAG(IS_CHROMEOS)
 void NetworkContext::LookupProxyAuthCredentials(
     const net::ProxyServer& proxy_server,
diff --git a/services/network/network_context.h b/services/network/network_context.h
index dad6a445ee3f6a0eb3d55d29fae643bde5bf706d..7040ef9f5a70d27ee329ff4f0ee715479bd2e650 100644
--- a/services/network/network_context.h
+++ b/services/network/network_context.h
@@ -536,7 +536,12 @@ class COMPONENT_EXPORT(NETWORK_SERVICE) NetworkContext
       AddAuthCacheEntryCallback callback) override;
   void SetCorsNonWildcardRequestHeadersSupport(bool value) override;
   void SetDohFallbackUpgradeAllowed(bool allowed) override;
-
+  // TODO(mmenke): Rename this method and update Mojo docs to make it clear this
+  // doesn't give proxy auth credentials.
+  void LookupServerBasicAuthCredentials(
+      const GURL& url,
+      const net::NetworkAnonymizationKey& network_anonymization_key,
+      LookupServerBasicAuthCredentialsCallback callback) override;
 #if BUILDFLAG(IS_CHROMEOS)
   void LookupProxyAuthCredentials(
       const net::ProxyServer& proxy_server,
diff --git a/services/network/public/mojom/network_context.mojom b/services/network/public/mojom/network_context.mojom
index 7493bd8dd37b36a3e7a96b1373619fee0b6a9d8e..d7c45d2fedbb5c96baa52e67b47467bfd467485c 100644
--- a/services/network/public/mojom/network_context.mojom
+++ b/services/network/public/mojom/network_context.mojom
@@ -1689,6 +1689,14 @@ interface NetworkContext {
   // supported in this network context.
   SetCorsNonWildcardRequestHeadersSupport(bool value);
 
+  // Looks up credentials in the HttpAuthCache using the origin and path from
+  // |url|. Only supports basic auth scheme. Only looks up server (not proxy)
+  // auth credentials, and only those that are usable in the scope of
+  // |network_anonymization_key|.
+  LookupServerBasicAuthCredentials(
+      url.mojom.Url url, NetworkAnonymizationKey network_anonymization_key)
+      => (AuthCredentials? credentials);
+
   // Looks up the proxy authentication credentials associated with
   // |proxy_server|, |auth_scheme| and |realm| in the HttpAuthCache.
   // |auth_scheme| is the authentication scheme of the challenge and it's
